lib/orbit/transformable.js

```diff
@@ -1,121 +1,118 @@
+    data: transform,
+    process: function() {
+      return processTransform.call(_this, transform);
+    }
+  });
+
+  return action.complete;
 }
 
-function queueTransformation(transformation) {
+function queueTransformResult(transform, result) {
   var _this = this;
 
-  var processor = this._transformationQueue.push({
-    data: transformation,
+  var action = this._transformQueue.push({
+    data: transform,
     process: function() {
-      return transformation.process();
+      return settleTransform.call(_this, transform, result);
     }
   });
 
-  return processor;
+  return action.complete;
 }
 
 var Transformable = {
   extend: function(object, actions) {
     if (object._transformable === undefined) {
       object._transformable = true;
-      object._transformationQueue = new ActionQueue();
+      object._transformQueue = new ActionQueue();
+      object._transformLog = {};
 
       Evented.extend(object);
 
-      object.didTransform = function(operation, inverse) {
-        var normalized = normalize(operation);
-        var transformation = transformationFor.call(this, normalized);
-        if (transformation) {
-          // console.log('Transformable#didTransform - matching transformation found', this.id, normalized, inverse);
-          transformation.pushCompletedOperation(normalized, inverse);
-
-        } else {
-          // console.log('Transformable#didTransform - createTransformation', this.id, normalized, inverse);
-          transformation = new Transformation(this);
-          transformation.pushCompletedOperation(normalized, inverse);
-          queueTransformation.call(this, transformation);
-        }
+      object.didTransform = function() {
+        Orbit.deprecate('`didTransform` has been deprecated. Please call `transformed` instead and supply a TransformResult as an argument.', true);
       };
 
-      object.currentTransformation = function() {
-        if (this._transformationQueue.current) return this._transformationQueue.current.data;
-      };
+      object.transformed = function(result) {
+        assert('Call `transformed` with an instanceof a TransformResult.', result instanceof TransformResult);
 
-      object.transform = function(operation) {
-        var normalized = normalize(operation);
-        var transformation = transformationFor.call(this, normalized);
-        var action;
+        var transform = new Transform(result.operations);
 
-        if (transformation) {
-          // console.log('transform - matching transformation found', this.id, normalized);
-          action = transformation.pushOperation(normalized);
+        // console.log('transformed - queued', this.id, transform, result);
+        return queueTransformResult.call(this, transform, result);
+      };
 
-          if (isArray(action)) {
-            return action[action.length - 1].complete;
-          } else {
-            return action.complete;
-          }
+      object.transform = function(transform) {
```

Comment:
I may be missing something but it looks like didTransform is always called as a direct result of calling transform(transform). In the case of orbit-firebase there's no direct connection between the call to transform(transform) and didTransform. The transform(transform) call would make changes to firebase, listeners attached to firebase would then capture any changes and transmit them through 'didTransform' (it remains to be determined how I tackle matching up the received changes with the original transform...).